### DISCUSSION TITLE
mcmap GUI

### DISCUSSION THREAD

**Discussion by spoutn1k** [spoutn1k/mcmap#63] [General]
Hi everyone,

I was told many times that a GUI would be a better way for people to use mcmap. A few weeks back #61 was created with source code for an old windows GUI. Updating this seemed like a good idea at the time, but taking a step back this code is OS-locked, and not very modern.

I am at a new chapter in my life and need to focus on work and things outside of programming for a while. I picked up mcmap as a out-of-date tool and now that it is fully-functionning, and performs almost as well as the original with support for new versions of Minecraft, I feel like my work is complete. While I won't drop support for mcmap anytime soon, I may not be able to dive as much as I would like on GUIs.

This is an interest check. I would like to know how many people are actually interested in such an app, and what platform they hope it to run on. A cross-platform tool would be the ideal choice, as I know technologies that appeared since the beginning of the project allow it.

If anyone with UI skill is interested in helping to make such a tool, I would be more than happy to explain and adapt mcmap for the needs of a front-end.

**Comment by jamaa** [comment]
I would be interested in a GUI (for Windows)!

**Comment by spoutn1k** [comment]
You asked for it, I took half a year to make it. The attached folder has a Qt interface to `mcmap`, tested on windows 10. This was a major pain to get to work on windows, but it finally renders properly. I am looking for feedback !

[Check it out in the releases](https://github.com/spoutn1k/mcmap/releases)

**Reply by jamaa** [reply]
It works! Thanks a lot! Next step: add a map preview for selecting the map area to be rendered :-) I like that the GUI automatically sets max. coordinates when you select a world, though.

**Reply by spoutn1k** [reply]
Whoah that is gonna be on the back of the back burner for now, my plan is to GPU-accelerate the code and the CLI interface before that. I'm way out of my depth making UIs, so it may come but expect a small delay™.

**Comment by luisl20** [comment]
Hi, i'm trying to use the GUI, but i don't know how i'm supposed to create the output. it just doesn't create any file that i can see after rendering. the only thing i didn't select was the colorfile because i don't have any

**Reply by spoutn1k** [reply]
Weird, it might be crashing, or the output might be empty. What version of the GUI are you using ? What minecraft version/map ? Please create an issue, so that the solution we find can help future users !

**Reply by luisl20** [reply]
i don't know which version of the gui it was, since the zip didn't have any reference to the version. The map was from 1.18.2, altough it was originally a 1.16.4 map. I've just tried using the 3.0.1 version of the gui, and it worked perfectly and actually rendered.
